src/software/geom/linear_spline2d_test.cpp

```diff
@@ -117,3 +117,46 @@ TEST(LinearSpline2dTest, test_get_knot_parametrization_values)
 
     EXPECT_EQ(std::vector<double>({0, 0.5, 1}), s.getKnotParametrizationValues());
 }
+
+TEST(LinearSpline2dTest, test_get_spline_segments)
+{
+    // linear spline with three points & two segments
+    {
+        LinearSpline2d spline({Point(1, 2), Point(2, 3), Point(0, -1)});
+        std::vector<SplineSegment2d> spline_segments = spline.getSplineSegments();
+        EXPECT_EQ(spline_segments.size(), 2);
+
+        Polynomial1d segment1_x({{0, 1}, {1.0 / 2.0, 2}});
+        Polynomial1d segment1_y({{0, 2}, {1.0 / 2.0, 3}});
+        Polynomial2d segment1(segment1_x, segment1_y);
+        EXPECT_EQ(spline_segments[0].getPolynomial(), segment1);
+        EXPECT_EQ(spline_segments[0].getParametrizationStartVal(), 0);
+        EXPECT_EQ(spline_segments[0].getParametrizationEndVal(), 1.0 / 2.0);
+
+        Polynomial1d segment2_x({{1.0 / 2.0, 2}, {1, 0}});
+        Polynomial1d segment2_y({{1.0 / 2.0, 3}, {1, -1}});
+        Polynomial2d segment2(segment2_x, segment2_y);
+        EXPECT_EQ(spline_segments[1].getPolynomial(), segment2);
+        EXPECT_EQ(spline_segments[1].getParametrizationStartVal(), 1.0 / 2.0);
+        EXPECT_EQ(spline_segments[1].getParametrizationEndVal(), 1);
+    }
+    // linear spline with two points and one segment
+    {
+        LinearSpline2d spline({Point(4.5, 3.2), Point(1.1, 3.34)});
+        std::vector<SplineSegment2d> spline_segments = spline.getSplineSegments();
+        EXPECT_EQ(spline_segments.size(), 1);
+
+        Polynomial1d segment1_x({{0, 4.5}, {1, 1.1}});
+        Polynomial1d segment1_y({{0, 3.2}, {1, 3.34}});
+        Polynomial2d segment1(segment1_x, segment1_y);
+        EXPECT_EQ(spline_segments[0].getPolynomial(), segment1);
+        EXPECT_EQ(spline_segments[0].getParametrizationStartVal(), 0);
+        EXPECT_EQ(spline_segments[0].getParametrizationEndVal(), 1);
+    }
+    // linear spline with one point & zero segments
+    {
+        LinearSpline2d spline({Point(6, 7)});
+        std::vector<SplineSegment2d> spline_segments = spline.getSplineSegments();
+        EXPECT_EQ(spline_segments.size(), 0);
+    }
+}
```

Comment:
I would suggest splitting this into three separate tests with descriptive names (e.g. `test_spline_with_three_points_has_two_segments`) so each case is isolated and failures are easier to trace.